packages/core/src/components/dialog/dialog.tsx

```diff
@@ -65,6 +65,11 @@ export interface DialogProps extends OverlayableProps, BackdropProps, Props {
      */
     isCloseButtonShown?: boolean;
 
+    /**
+     * @default "dialog"
+     */
+    role?: React.AriaRole;
```

Comment:
Could we have an explicit list of values we expect to be valid given the setup of this component? `"dialog" | "alertdialog"` feels like a reasonable place to start that can be expanded as needed.

Reply:
https://github.com/palantir/blueprint/pull/6846/commits/97104bbe7054b687d62fd36107e2448147f12742